### PR TITLE
Model tweaks

### PR DESCRIPTION
**Pull request by averytribbett**
#49 

1. Created a connection between PriceHistorySchema and CardSchema.
2. Changed day, month, and year to just date in both the PriceHistorySchema and PriceHistory interface.

### CODE REVIEW

File: api/src/interfaces/priceHistory.interface.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import { Document } from 'mongoose';
 
 export interface PriceHistoryData extends Document {
-    month: number;
-    day: number; 
-    year: number; 
+    date: Date;
```
Comment (Kobe16) [inline api/src/interfaces/priceHistory.interface.ts:4]: I think we also used month + day + year in the frontend. Might be worth it to change that to the Date type in the frontend too.

File: api/src/models/priceHistory.model.ts [hunk 2]
```diff
@@ -22,6 +23,12 @@ const PriceHistorySchema: Schema = new Schema({
     type: Number, 
     required: true, 
   },
+  // Required because for a price history object to be created it must be linked to a card.
+  cardId: {
+    type: Schema.Types.ObjectId,
+    ref: PokemonCardModel,
+    required: true,
+  }
```
Comment (Kobe16) [inline api/src/models/priceHistory.model.ts:30]: I think that this looks good to me.

**Review by Kobe16 — CHANGES_REQUESTED** [review]

**Review by Kobe16 — APPROVED** [review]
Looks good 👍

File: api/src/models/priceHistory.model.ts [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-    type: Number, 
-    required: true, 
-  }, 
-  year: {
-    type: Number, 
-    required: true, 
-  }, 
+  // Not sure if id is required 
+  // I think it is because the cards references PriceHistorySchema
+  id: {
```
Comment (jbrw1984) [inline api/src/models/priceHistory.model.ts:8]: You may not need this as it'll be automatically added with Mongoose. You should check the documentation.

**Review by jbrw1984 — APPROVED** [review]
Comments, but approving.

**Pull request merged**